libs/gpu-codec/src/main/java/org/elasticsearch/gpu/codec/DelegatingFlatVectorsScorer.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.gpu.codec;
+
+import org.apache.lucene.codecs.hnsw.FlatVectorsScorer;
+import org.apache.lucene.index.KnnVectorValues;
+import org.apache.lucene.index.VectorSimilarityFunction;
+import org.apache.lucene.util.hnsw.RandomVectorScorer;
+import org.apache.lucene.util.hnsw.RandomVectorScorerSupplier;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Delegates all operations to a given delegate, while exposing the most recently seen KnnVectorValues.
+ */
+final class DelegatingFlatVectorsScorer implements FlatVectorsScorer {
+
+    private final FlatVectorsScorer delegate;
+    private KnnVectorValues lastVectorValues;
+
+    DelegatingFlatVectorsScorer(FlatVectorsScorer delegate) {
+        this.delegate = Objects.requireNonNull(delegate);
+    }
+
+    public KnnVectorValues getLastVectorValuesOrNull() {
+        return lastVectorValues;
```

Comment:
Can the same instance of DelegatingFlatVectorsScorer be accessed by multiple threads at the same time?

If yes, should we guard  access of `lastVectorValues` for that?